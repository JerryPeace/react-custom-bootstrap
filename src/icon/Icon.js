import PropTypes from 'prop-types';
import React from 'react';

class Icon extends React.Component {
    static propTypes = {
        /**
         * icon type
         */
        src: PropTypes.string.isRequired
    };

    render() {
        return (
            <img className='tm-button-icon' src={this.props.src}>
                {this.props.children}
            </img>
        );
    }
}


export default Icon;
