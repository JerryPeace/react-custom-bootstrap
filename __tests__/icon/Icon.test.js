import React from 'react';
import { mount } from 'enzyme';
import Icon from 'icon/Icon';

describe('Icon', () => {
    test('render Icon correctly', () => {
        const icon = mount(<Icon src='ddd.jpg'/>);
        expect(icon).toMatchSnapshot();
    });
});
