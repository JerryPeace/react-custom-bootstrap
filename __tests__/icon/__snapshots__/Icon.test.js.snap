// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon render Icon correctly 1`] = `
<Icon
  src="ddd.jpg"
>
  <img
    className="tm-button-icon"
    src="ddd.jpg"
  />
</Icon>
`;
